Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -151,6 +154,50 @@ void AODProducerWorkflowDPL::createCTPReadout(const o2::globaltracking::RecoCont
     }
   }
   LOG(info) << "# of TRD bogus triggers:" << cntwarnings;
+
+  // EMCAL:
+  cntwarnings = 0;
+  orbitPrev = 0;
+  bcPrev = 0;
+  // Get class mask for EMCAL:
+  uint64_t classMaskEMCAL = 0;
+  for (const auto& trgclass : ctpcfg->getCTPClasses()) {
+    if (trgclass.cluster->getClusterDetNames().find("EMC")) {
+      classMaskEMCAL = trgclass.classMask;
+    }
+  }
+  for (auto& emcrec : caloEMCCellsTRGR) {
+    auto orbitPrevT = orbitPrev;
+    auto bcPrevT = bcPrev;
+    bcPrev = emcrec.getBCData().bc;
+    orbitPrev = emcrec.getBCData().orbit;
+    if (orbitPrev < orbitPrevT || bcPrev >= o2::constants::lhc::LHCMaxBunches || (orbitPrev == orbitPrevT && bcPrev < bcPrevT)) {
```

Comment:
how is this possible?

Reply:
What I had (still have) in mind is to combine all 3 triggers into a single lambda. From what I read in the other PR TRD needs this, and for EMCAL and PHOS it should be not needed but also hurt, as commented in the code, since I am not aware of spurious triggers sent by us. I kept it there in the preparation of merging.